packages/lib/passport-interface/test/SubscriptionManager.spec.ts

```diff
@@ -137,6 +199,146 @@ describe("Subscription Manager", async function () {
+        signatureType: "sempahore-signature-pcd"
+      }
+    };
+
+    const sub = await manager.subscribe(providerUrl, feed, undefined);
+
+    const action = {
+      type: PCDActionType.ReplaceInFolder,
+      folder: "TEST",
+      pcds: [
+        await SemaphoreIdentityPCDPackage.serialize(
+          await SemaphoreIdentityPCDPackage.prove({
+            identity: new Identity()
+          })
+        )
+      ]
+    };
+
+    manager.setError("1", {
+      type: SubscriptionErrorType.PermissionError,
+      actions: [action]
+    });
+
+    const cleanupListener = manager.updatedEmitter.listen(() => {
+      assert.fail("Listener should never be called.");
+    });
+
+    expect(manager.getProviders()).to.have.length(1);
+    expect(manager.getActiveSubscriptions()).to.have.length(1);
+    expect(manager.getSubscription(sub.id)).to.not.be.undefined;
+
+    const clone = manager.clone();
+
+    expect(clone.getProviders()).to.deep.eq(manager.getProviders());
+    expect(clone.getSubscriptionsByProvider()).to.deep.eq(
+      manager.getSubscriptionsByProvider()
+    );
+    expectSameSubs(manager, clone);
+    expect(clone.getAllErrors()).to.deep.eq(manager.getAllErrors());
+
+    clone.unsubscribe(sub.id); // Would trigger listener if it were on the clone.
+    cleanupListener();
+  });
+
+  it("merging unique data should work", async function () {
+    const entries1 = [
+      { provider: "1", feed: "1A" },
+      { provider: "1", feed: "1B" },
+      { provider: "2", feed: "2A" }
+    ];
+    const mgr1 = setupTestManager(entries1);
+    const entries2 = [
+      { provider: "1", feed: "1C" },
+      { provider: "B", feed: "B1" },
+      { provider: "C", feed: "C1" },
+      { provider: "C", feed: "2A" }
+    ];
+    const mgr2 = setupTestManager(entries2);
+    const allSubs = [...entries1, ...entries2];
+
+    expect(mgr1.merge(mgr2)).to.deep.eq({
+      newProviders: 2,
+      newSubscriptions: 4
+    });
+    expect(mgr1.getActiveSubscriptions()).to.have.length(7);
+    expectAllTestSubs(mgr1, allSubs);
+
+    expect(mgr2.merge(mgr1)).to.deep.eq({
+      newProviders: 1,
+      newSubscriptions: 3
+    });
+    expect(mgr2.getActiveSubscriptions()).to.have.length(7);
+    expectAllTestSubs(mgr2, allSubs);
+
+    expectSameSubs(mgr1, mgr2);
+  });
+
+  it("merging should ignore duplicates", async function () {
+    const mgr = setupTestManager([
+      { provider: "1", feed: "1A" },
+      { provider: "1", feed: "1B" },
+      { provider: "2", feed: "2A" }
+    ]);
+    expect(mgr.getActiveSubscriptions()).to.have.length(3);
+
+    // Self-merge should be a nop.
+    expect(mgr.merge(mgr)).to.deep.eq({ newProviders: 0, newSubscriptions: 0 });
+    expect(mgr.getActiveSubscriptions()).to.have.length(3);
+
+    // Merge wth a clone should also be a nop.
+    expect(mgr.merge(mgr.clone())).to.deep.eq({
+      newProviders: 0,
+      newSubscriptions: 0
+    });
+    expect(mgr.getActiveSubscriptions()).to.have.length(3);
```

Comment:
seems a bit superfluous given that you test `clone` above

Reply:
I think this is a test of merge(), confirming that it handles 2 identical collections correctly, and doesn't get confused by the shared objects of the shallow clone.